pandas/tests/io/excel/test_readers.py

```diff
@@ -1143,3 +1143,22 @@ def test_header_with_index_col(self, engine, filename):
             filename, sheet_name="Sheet1", index_col=0, header=[0, 1]
         )
         tm.assert_frame_equal(expected, result)
+
+    def test_read_datetime_multiindex(self, engine, read_ext):
+        # GH 34748
+        if engine == "pyxlsb":
+            pytest.xfail("Sheets containing datetimes not supported by pyxlsb")
+
+        f = "test_datetime_mi" + read_ext
+        with pd.ExcelFile(f) as excel:
+            actual = pd.read_excel(excel, header=[0, 1], index_col=0, engine=engine)
+        expected_column_index = pd.MultiIndex.from_tuples(
+            [(pd.to_datetime("02/29/2020"), pd.to_datetime("03/01/2020"))],
+            names=[
+                pd.to_datetime("02/29/2020").to_pydatetime(),
```

Comment:
Are the to_pydatetime calls required?

Reply:
Because when the Excel reader creates the names of the index, the types are of `dt.datetime` not the pandas `datetime`

Reply:
hmm, where is this done? this is unfortunately as these should actually be Timestamp

Reply:
> hmm, where is this done? this is unfortunately as these should actually be Timestamp

So is this a separate issue - that we don't want the *names* to be dt.datetime ?  If so, I will create an issue for that.